app/code/Magento/InventoryConfigurableProduct/Plugin/Block/AdaptMatrix.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryConfigurableProduct\Plugin\Block;
+
+use Magento\ConfigurableProduct\Block\Adminhtml\Product\Edit\Tab\Variations\Config\Matrix;
+use Magento\InventoryCatalog\Model\IsSingleSourceModeInterface;
+use Magento\InventoryConfigurableProduct\Model\GetQuantityInformationPerSource;
+
+/**
+ * Add value for field "quantityPerSource" for grid "Associated Products" and "Disassociated Products"
+ * on step "Summary".
+ */
+class AdaptMatrix
+{
+    /**
+     * @var GetQuantityInformationPerSource
+     */
+    private $getQuantityInformationPerSource;
+
+    /**
+     * @var IsSingleSourceModeInterface
+     */
+    private $isSingleSourceMode;
+
+    /**
+     * @param GetQuantityInformationPerSource $getQuantityInformationPerSource
+     * @param IsSingleSourceModeInterface $isSingleSourceMode
+     */
+    public function __construct(
+        GetQuantityInformationPerSource $getQuantityInformationPerSource,
+        IsSingleSourceModeInterface $isSingleSourceMode
+    ) {
+        $this->getQuantityInformationPerSource = $getQuantityInformationPerSource;
+        $this->isSingleSourceMode = $isSingleSourceMode;
+    }
+
+    /**
+     * @param Matrix $subject
+     * @param $result
+     *
+     * @return array
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterGetProductMatrix(
+        Matrix $subject,
+        $result
+    ): array {
+        if ($this->isSingleSourceMode->execute() === false) {
+            foreach ($result as $key => $variation) {
```

Comment:
According to the base implementation, the result can be `null`
SO you will get PHP Warning `: Invalid argument supplied for foreach()`

Reply:
Fixed.